storage/gcsemu/gcsemu.go

```diff
@@ -122,6 +126,18 @@ func (g *GcsEmu) Handler(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if r.Header.Get("content-encoding") == "gzip" {
+		gzipReader, err := gzip.NewReader(r.Body)
+		if err != nil && io.EOF != err {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
This should probably be BadRequest right, if the client sent a payload that's not valid gzip?

Reply:
removed